cmake/find/FindFilesystem.cmake

```diff
@@ -236,13 +240,40 @@ if (NOT _found AND "Boost" IN_LIST want_components)
     find_package(Boost COMPONENTS filesystem)
     if (TARGET Boost::filesystem)
         set(_found TRUE)
-        set(CXX_FILESYSTEM_HEADER boost/filesystem.hpp CACHE STRING "The header that should be included to obtain the filesystem APIs")
-        set(CXX_FILESYSTEM_NAMESPACE boost::filesystem CACHE STRING "The C++ namespace that contains the filesystem APIs")
+        # Bug fix by Bryan Tan
+        set(CXX_FILESYSTEM_HEADER boost/filesystem.hpp CACHE STRING "The header that should be included to obtain the filesystem APIs" FORCE)
+        set(CXX_FILESYSTEM_NAMESPACE boost::filesystem CACHE STRING "The C++ namespace that contains the filesystem APIs" FORCE)
+        # End bug fix
         add_library(std_filesystem INTERFACE)
         target_compile_features(std_filesystem INTERFACE cxx_std_17)
         target_link_libraries(std_filesystem INTERFACE Boost::boost Boost::filesystem)
+        target_include_directories(std_filesystem INTERFACE ${Boost_INCLUDE_DIRS})
```

Comment:
This line should be obsolete, since it is part of the target Boost::filesystem and Boost::boost. Or is this not working, since we have an Interface of an Interface library?

Reply:
I will have to check this later, because my first attempt at making it compile with Boost needed this line.

Reply:
Turns out to be unnecessary.